pandas/core/ops.py

```diff
@@ -42,6 +42,47 @@
     ABCSparseSeries, ABCSparseArray)
 
 
+# -----------------------------------------------------------------------------
+# Ops Wrapping Utilities
+
+def get_op_result_name(left, right):
+    """
+    Find the appropriate name to pin to an operation result.  This result
+    should always be either an Index or a Series.
+
+    Parameters
+    ----------
+    left : {Series, Index}
+    right : object
+
+    Returns
+    -------
+    name : object
+        Usually a string
+    """
+    # `left` is always a pd.Series when called from within ops
+    if isinstance(right, (ABCSeries, pd.Index)):
+        name = _maybe_match_name(left, right)
+    else:
+        name = left.name
+    return name
+
+
+def _maybe_match_name(a, b):
```

Comment:
do we need this as a standalone? (IOW can you replace the current usage with get_op_result_name)?

Reply:
if not can you doc-string this

Reply:
A lot of the uses can be replaced without affecting any existing behavior.  I'll see if I can get them all in a way that doesn't necessitate new tests.  If not I'll kill it off in the next pass.

Reply:
Yep, all non-test usages can be replaced.  I've done that and then also added a docstring.  Getting rid of it and updating the tests to test get_op_result_name directly will wait for another round with narrower scope.

Reply:
ok looks good. next pass let's try to remove this (I think you did, just need to transfer the tests)